### PR TITLE
test(pager): refactor tests to ts

### PR DESCRIPTION
**Pull request by DlgSHi**
### Proposed behaviour

- Rename the `pager.cy.js` to `pager.cy.tsx` file in `./cypress/components/pager/` folder.
- Refactor tests to Typescript.  

### Current behaviour

Pager tests are in js not ts.

### Checklist

- [x] Commits follow our style guide
- [ ] Related issues linked in commit messages if required
- [ ] Screenshots are included in the PR if useful
- [ ] All themes are supported if required
- [ ] Unit tests added or updated if required
- [x] Cypress automation tests added or updated if required
- [ ] Storybook added or updated if required
- [ ] Translations added or updated (including creating or amending translation keys table in storybook) if required
- [ ] Typescript `d.ts` file added or updated if required
- [ ] Related docs have been updated if required

#### QA

- [x] Tested in CodeSandbox/storybook
- [x] Add new Cypress test coverage if required
- [x] Carbon implementation matches Design System/designs
- [ ] UI Tests GitHub check reviewed if required

### Testing instructions
- [x] Run `npx cypress open --component` to check if the `pager.cy.tsx` file passed
- [x] Run `npx cypress run --component` to check none of the other `*.cy.tsx` files have regressed

### CODE REVIEW

**Review by divyajindel — APPROVED** [review]

File: cypress/components/pager/pager.cy.tsx [hunk 1]
```diff
@@ -297,7 +299,7 @@ context("Test for Pager component", () => {
   });
 
   describe("check funtionality for Pager component", () => {
-    it.each([[-1], [-10], [-100], testData[0], testData[1]])(
+    it.each([-1, -10, -100, testData[0], testData[1]])(
```
Comment (edleeks87) [inline cypress/components/pager/pager.cy.tsx:302]: suggestion: you could spread the testData array here but it's not a big deal

```suggestion
    it.each([-1, -10, -100, ...testData)(
```

File: cypress/components/pager/pager.cy.tsx [hunk 2]
```diff
@@ -397,121 +395,100 @@ context("Test for Pager component", () => {
...
-        .then(() => {
-          // eslint-disable-next-line no-unused-expressions
-          expect(callback).to.have.been.calledOnce;
-        });
+      nextArrow().click();
+      cy.get("@onNext").should("have.been.calledOnce");
     });
 
-    it.each([["Enter"], ["Space"]])(
+    it.each([keysToTrigger[0], keysToTrigger[1]])(
```
Comment (edleeks87) [inline cypress/components/pager/pager.cy.tsx:412]: suggestion: you could just pass the array here like so, again not a big deal

```suggestion
    it.each(keysToTrigger)(
```

**Review by ZhuoyuJin — APPROVED** [review]

File: cypress/components/pager/pager.cy.tsx [hunk 3]
```diff
@@ -397,121 +395,100 @@ context("Test for Pager component", () => {
...
-        .then(() => {
-          // eslint-disable-next-line no-unused-expressions
-          expect(callback).to.have.been.calledOnce;
-        });
+      previousArrow().click();
+      cy.get("@onPrevious").should("have.been.calledOnce");
     });
 
-    it.each([["Enter"], ["Space"]])(
+    it.each([keysToTrigger[0], keysToTrigger[1]])(
```
Comment (edleeks87) [inline cypress/components/pager/pager.cy.tsx:435]: Same as above

File: cypress/components/pager/pager.cy.tsx [hunk 4]
```diff
@@ -397,121 +395,100 @@ context("Test for Pager component", () => {
...
-        .then(() => {
-          // eslint-disable-next-line no-unused-expressions
-          expect(callback).to.have.been.calledOnce;
-        });
+      firstArrow().click();
+      cy.get("@onFirst").should("have.been.calledOnce");
     });
 
-    it.each([["Enter"], ["Space"]])(
+    it.each([keysToTrigger[0], keysToTrigger[1]])(
```
Comment (edleeks87) [inline cypress/components/pager/pager.cy.tsx:460]: Same as previous comment

**Review by edleeks87 — APPROVED** [review]

**Review by stephenogorman — APPROVED** [review]

**Pull request merged**